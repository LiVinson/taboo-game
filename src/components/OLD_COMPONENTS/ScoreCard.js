import React from "react"

export default function ScoreCard({ score1, score2 }) {
  return (
    <div>
      <p>
        Team 1:<span>{score1}</span>
      </p>
      <p>
        Team 2:<span>{score2}</span>
      </p>
    </div>
  )
}
