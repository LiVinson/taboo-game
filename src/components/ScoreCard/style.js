import styled from 'styled-components'

export const Scores = styled.div`
	font-family: ${(props) => props.theme.text.display};
	font-size: 2rem;
	color: ${(props) => props.theme.color.grayDark2};
	text-align: center;
	/* adds alpha to lighten the color without impacting children opacity */
	background-color: ${(props) => props.theme.color.offWhite}DD;
	border: solid 2px ${(props) => props.theme.color.accent3};
	border-radius: 3px;
	margin-bottom: 2rem;
	padding: 1rem 0rem;
`

export const ScoreTitle = styled.p`
	font-size: 2.5rem;
	color: ${(props) => props.theme.color.accent3};
	&:after {
		display: block;
		content: '';
		width: 80%;
		height: 1.5px;
		margin: 0.5rem auto;
		background-color: ${(props) => props.theme.color.lightGray2};
	}
`
export const Score = styled.li`	
		margin: 0.8rem auto 1rem auto;
	}
`
