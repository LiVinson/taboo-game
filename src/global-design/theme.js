const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
