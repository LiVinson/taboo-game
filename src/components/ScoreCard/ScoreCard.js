import React from 'react'
import PropTypes from "prop-types"
import { Scores, ScoreTitle, Score } from './style'
import  List  from 'components/shared/List'

const ScoreCard = ({ teamScores }) => {
	return (
		<Scores>
			<ScoreTitle>Scores</ScoreTitle>
			<List>
				{teamScores.map((score, index) => (
					<Score key={index}>
						Team {index + 1}: <span>{score}</span>
					</Score>
				))}
			</List>
		</Scores>
	)
}

ScoreCard.propTypes = {
	teamScores: PropTypes.array.isRequired
}
export default ScoreCard
