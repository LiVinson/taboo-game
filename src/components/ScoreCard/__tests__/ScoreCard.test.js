import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import ScoreCard from "../ScoreCard"

test("ScoreCard renders correctly", () => {
    const props = {
        teamScores: [12, 10]
    }

    const wrapper = renderer.create(
        <ThemeProvider theme={theme}>
            <ScoreCard {...props} />
        </ThemeProvider>
    ).toJSON()

    expect(wrapper).toMatchSnapshot()
})