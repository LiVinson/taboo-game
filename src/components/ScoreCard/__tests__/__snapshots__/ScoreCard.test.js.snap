// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreCard renders correctly 1`] = `
<div
  className="sc-AxjAm fdAzhr"
>
  <p
    className="sc-AxirZ ibSLEF"
  >
    Scores
  </p>
  <ul
    className="sc-AxhCb iSuNyo"
  >
    <li
      className="sc-AxiKw gPyhNF"
    >
      Team 
      1
      : 
      <span>
        12
      </span>
    </li>
    <li
      className="sc-AxiKw gPyhNF"
    >
      Team 
      2
      : 
      <span>
        10
      </span>
    </li>
  </ul>
</div>
`;
